pymc3/distributions/discrete.py

```diff
@@ -603,6 +603,98 @@ def logcdf(value, mu):
         )
 
 
+class GeneralizedPoissonRV(RandomVariable):
+    name = "generalized_poisson"
+    ndim_supp = 0
+    ndims_params = [0, 0]
+    dtype = "int64"
+    _print_name = ("GeneralizedPoisson", "\\operatorname{GeneralizedPoisson}")
+
+    @classmethod
+    def rng_fn(cls, rng, theta, lam, size):
```

Comment:
Is there an expected shape for these parameters? This implementation is assuming that `theta` is an array and `lam` should be a scalar, so the method will return an array of random samples, one for each value of `theta`.

Reply:
The minimum expected shape is the one you define in ndims_supp, which as a `[0,0]` means two scalar values. This sounds correct for this univariate distribution.

Can you tweak the logic slightly to allow for two scalar arguments as well as the vector case?

It also seems like you generating code could be vectorized (instead of looping through the `i`) for efficiency

Reply:
Updated the logic so that both arguments can be either scalars or vectors, and vectorized the generating code.